### PR TITLE
MI300 Support

### PR DESCRIPTION
**Pull request by msaudulhassan**
Enable MI300 support.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -141,7 +141,7 @@ RUN git submodule update --init --recursive
 ARG FIN_DIR=$MIOPEN_DIR/fin
 WORKDIR $FIN_DIR
 # Can be a branch or a SHA
-ARG FIN_BRANCH=
```
Comment (cderb) [inline Dockerfile]: Please open a PR for this branch in MIFin. We'll keep the branch here on fin's develop.

Comment (msaudulhassan) [reply Dockerfile]: https://github.com/ROCmSoftwarePlatform/MIFin/pull/87

**Review by cderb — APPROVED** [review]
LGTM

**Review by reidkwja — APPROVED** [review]
looks fine

**Pull request merged**